demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1969 @@
+    pxOtaInterfaces->os.mem.malloc = Malloc_FreeRTOS;
+    pxOtaInterfaces->os.mem.free = Free_FreeRTOS;
+
+    /* Initialize the OTA library MQTT Interface.*/
+    pxOtaInterfaces->mqtt.subscribe = prvMqttSubscribe;
+    pxOtaInterfaces->mqtt.publish = prvMqttPublish;
+    pxOtaInterfaces->mqtt.unsubscribe = prvMqttUnSubscribe;
+
+    /* Initialize the OTA library PAL Interface.*/
+    pxOtaInterfaces->pal.getPlatformImageState = otaPal_GetPlatformImageState;
+    pxOtaInterfaces->pal.setPlatformImageState = otaPal_SetPlatformImageState;
+    pxOtaInterfaces->pal.writeBlock = otaPal_WriteBlock;
+    pxOtaInterfaces->pal.activate = otaPal_ActivateNewImage;
+    pxOtaInterfaces->pal.closeFile = otaPal_CloseFile;
+    pxOtaInterfaces->pal.reset = otaPal_ResetDevice;
+    pxOtaInterfaces->pal.abort = otaPal_Abort;
+    pxOtaInterfaces->pal.createFile = otaPal_CreateFileForRx;
+}
+/*-----------------------------------------------------------*/
+
+static void prvOTAAgentTask( void * pParam )
+{
+    /* Calling OTA agent task. */
+    OTA_EventProcessingTask( pParam );
+    LogInfo( ( "OTA Agent stopped." ) );
+
+    vTaskDelete( NULL );
+}
+/*-----------------------------------------------------------*/
+
+static void prvMQTTAgentTask( void * pParam )
+{
+    BaseType_t xResult = pdFAIL;
+    MQTTStatus_t xMQTTStatus = MQTTSuccess;
+
+    ( void ) pParam;
+
+    do
+    {
+        /* MQTTAgent_CommandLoop() is effectively the agent implementation.  It
+         * will manage the MQTT protocol until such time that an error occurs,
+         * which could be a disconnect.  If an error occurs the MQTT error code
+         * is returned and the queue left uncleared so there can be an attempt to
+         * clean up and reconnect however the application writer prefers. */
+        xMQTTStatus = MQTTAgent_CommandLoop( &xGlobalMqttAgentContext );
+
+        /* Clear Agent queue so that any pending MQTT operations are not processed. */
+        MQTTAgent_CancelAll( &xGlobalMqttAgentContext );
+
+        /* Success is returned for application intiated disconnect or termination.
+         * The socket will also be disconnected by the caller. */
+        if( xMQTTStatus != MQTTSuccess )
+        {
+            xResult = prvSuspendOTA();
+            configASSERT( xResult == pdPASS );
+
+            LogInfo( ( "Suspended OTA agent." ) );
+
+            /* Reconnect TCP. */
+            ( void ) SecureSocketsTransport_Disconnect( &xNetworkContextMqtt );
+
+            xResult = prvConnectToMQTTBroker();
+            configASSERT( xResult == pdPASS );
+
+            xResult = prvResumeOTA();
+            configASSERT( xResult == pdPASS );
+
+            LogInfo( ( "Resumed OTA agent." ) );
+        }
+    } while( xMQTTStatus != MQTTSuccess );
+
+    vTaskDelete( NULL );
+}
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvSuspendOTA( void )
+{
+    /* OTA library return status. */
+    OtaErr_t xOtaError = OtaErrNone;
```

Comment:
maintainability: status and error codes should always be initialized to values representing error/fail rather than success - both `xOtaError` and `xStatus`.

Reply:
Thanks, Updated in https://github.com/aws/amazon-freertos/pull/3188/commits/4a4f10bee521486d849ca1ca0f93e90f266ca035